lib/galaxy/datatypes/gsas2.py

```diff
@@ -0,0 +1,117 @@
+"""
+classes of datatypes from GSASII
+"""
+
+import logging
+
+from galaxy.datatypes.binary import Binary
+from galaxy.datatypes.data import Text
+
+log = logging.getLogger(__name__)
+
+
+class Gsas2Project(Binary):
+    """
+    GSASII .gpx project file format
+    file is pickled can be read by pickle.load(fp,encoding='latin-1')
+    """
+
+    file_ext = "gpx"
+    # sniffing disabled for now
+
+    def sniff(self, filename):
+        """
+        Returns false and the user must manually set
+        """
+        return False
+
+
+class Prm(Text):
+    """
+    GSASII .prm instrument parameter format
+    """
+
+    file_ext = "prm"
+
+    def sniff(self, filename):
+        """
+        Try to guess if the file is a GSASII .prm isntrument parameter file
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('gsas_instrument.prm')
+        >>> InstrumentParameter().sniff(fname)
+        True
+
+        >>> fname = get_test_fname('gsas_instrument.instprm')
+        >>> InstrumentParameter().sniff(fname)
+        False
+        """
+
+        header = open(filename).read(22)
+        return header == "            1234567890"
+
+
+class Instprm(Text):
+    """
+    GSASII .instprm instrument parameter format
+    """
+
+    file_ext = "instprm"
+
+    def sniff(self, filename):
+        """
+        Try to guess if the file is a GSASII .instprm isntrument parameter file
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('gsas_instrument.instprm')
+        >>> InstrumentParameter2().sniff(fname)
+        True
+
+        >>> fname = get_test_fname('gsas_instrument.prm')
+        >>> InstrumentParameter2().sniff(fname)
```

Comment:
```suggestion
        >>> Instprm().sniff(fname)
```
Need to update test for new classname.